### PR TITLE
LPS-192074[Marketplace] Field validation refactoring

### PR DESCRIPTION
**Pull request by juliocezar96**
[LPS-192074](https://liferay.atlassian.net/browse/LPS-192074)

### CODE REVIEW

**Comment by juliocezar96** [comment]
## :fire: GitRay Report 
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
 ## Jira Tasks 

 <html><details><summary>More details about: <b>LPS-192074 [FE] Field validation refactoring</b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-192074">[FE] Field validation refactoring</a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Low</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Progress</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Julio Cezar</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Julio Cezar</td>
        </tr>
                </table>
          </details></html> 

---

Hey, don't forget to update all related tickets with correct status, such as **In Review** :wink: 
### Git Stats 

3 files changed, 196 insertions(+), 61 deletions(-)

**Comment by juliocezar96** [comment]
ci:test:sf

**Review by joanagrego — APPROVED** [review]

**Comment by juliocezar96** [comment]
ci:test:relevant

**Comment by juliocezar96** [comment]
ci:test:relevant

**Comment by juliocezar96** [comment]
ci:test:relevant

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/schema/zod.ts [hunk 1]
```diff
@@ -32,6 +32,16 @@ const zodSchema = {
 			),
 	}),
 
+	invitedNewMember: z.object({
+		emailAddress: z
+			.string()
+			.nonempty('Please enter an email ')
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/schema/zod.ts:38]:
```suggestion
			.nonempty('Please enter an email')
```

**Review by claricebrasil — CHANGES_REQUESTED** [review]
Good job! I made a few suggestions just for improvement!!

**Comment by juliocezar96** [comment]
ci:test:relevant

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/InviteMemberModal/InviteMemberModal.tsx [hunk 2]
```diff
@@ -220,6 +290,23 @@ export function InviteMemberModal({
...
+						(accountRole) => roleCheck.roleName === accountRole.name
+					)
+			)
+			.map((roleCheck) => roleCheck.roleName);
+
+		if (!rolesChecked.length || !rolesSelected.length) {
+			setError('roles', {message: 'Please select at least one role'});
+		}
+
+		setValue('roles', rolesSelected as any);
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/InviteMemberModal/InviteMemberModal.tsx:308]: Hey @juliocezar96, I think you can create a function just to check if the roles are selected, filter these roles and list.
I was thinking something like this:

```
function isRoleSelected(roleCheck, accountRoles) {
return roleCheck.isChecked && (accountRoles?.some((accountRole) => roleCheck.roleName === accountRole.name) ?? false)};
```

Then you can filter and get the list like you did:
`const rolesSelected = rolesChecked.filter((roleCheck) => isRoleSelected(roleCheck, accountRoles)).map((roleCheck) => roleCheck.roleName)`

This way we can break the function to be more descriptive and readble. What do you think? Makes sense?

**Review by claricebrasil — CHANGES_REQUESTED** [review]

**Comment by juliocezar96** [comment]
ci:test:sf

**Review by claricebrasil — APPROVED** [review]
Great job!

**Comment by claricebrasil** [comment]
ci:forward

**Comment by juliocezar96** [comment]
ci:forward

**Comment by LuanaKSilva** [comment]
ci:stop

**Comment by juliocezar96** [comment]
#4575 resend

**Pull request closed without merging**